src/SUFST/inc/can_publisher.h

```diff
@@ -2,10 +2,21 @@
 #define CAN_PUBLISHER_H
 #include <stdint.h>
 #include <tx_api.h>
+#include "telemetry_protocol.h"
 
-extern TX_QUEUE queue;
-UINT queue_send_thread_create(TX_BYTE_POOL* stack_pool_ptr);
-void queue_send_thread_entry(ULONG input);
+#define TX_QUEUE_SIZE 10
+
+typedef struct {
+
+  TX_THREAD thread;
+
+  TX_QUEUE tx_queue;
 
+  rtcan_msg_t* tx_queue_mem[TX_QUEUE_SIZE];
+
+} publisher_context_t;
+
+UINT can_publisher_init(publisher_context_t* publisher_ptr, TX_BYTE_POOL* stack_pool_ptr);
+void queue_send_thread_entry(ULONG input);
```

Comment:
This function prototype isn't part of the module's public API, so I don't think it should be in the header. It should be moved to the start of the source file.